### DISCUSSION TITLE
Eye Gaze Bearing Values Questions

### DISCUSSION THREAD

**Discussion by desi-dimitrova** [vladmandic/human#252] [General]
Hi Vlad,

I have a couple of questions:

1. Can you help me understand when do we get negative values in eye gaze bearing, in which cases? 
2. If the eye gaze bearing is 0, am I right to think that the user is looking down? If not, would that mean the user is looking center?
3. When a user is looking up, the eye gaze bearing value should go above 0, correct?

**Comment by vladmandic** [comment]
you can see how its done in <https://github.com/vladmandic/human/blob/f7ecc00d90be1c8b4e2fb50c59c54e7b0ddbd778/src/face/angles.ts#L5>  
but simplified its:

1. calculate eye center from eye coordinates
2. calculate eye iris center position
3. compare iris position to what should be eye center
4. calculate iris bearing angle (in radians) using **arctan** function for difference of iris center to eye center on x-axis and y-axis
5. add PI/2 value so angle=90 is at center

actual angle should be 0 looking down to PI/2 looking straight to PI looking up (all in radians), but anything lower than 45 and higher than 135 degrees is unreliable is basically junk values.

in general, eye gaze bearing is not reliable as eye is a really small object to measure - calculating overall eye size and iris can be off by few pixels and it has massive impact.

**Reply by desi-dimitrova** [reply]
Thank you for the information! 

I was asking because I am looking for a way to detect if a person is looking at the lens but I noticed using eye gaze bearing does not work for everyone, so I am now going to explore a workaround in which I will use movenet to detect the center of the eyes and the ears and check what would be the angle between those points when looking down/up and at the lens. I feel it may not work but I will give it a try. If you have any suggestions how I can detect if a person is looking at the lens, please let me know. Thanks! :-)

**Comment by vladmandic** [comment]
eye gaze is not an accurate thing because of:
- eyes are relatively small, so looking at iris position inside eye is not precise. and being off by a pixel or two means a lot.
- eyes naturally get covered by eyelids, especially when looking down.

but if you're looking at **face** angle (that's what you're talking about when looking at eyes vs ears), `movenet` would be worst option since its precision of face objects is small - after all, its designed for full-body. not even looking at model details, just look at resolution - its 192px for full input which - and how big are eyes and ears in that?

anyhow, `human` already has much higher precision of face angles, just look at results for `rotation.angle` instead of `rotation.gaze`.
and math there is done using proper rotation matrix, take a look at: <https://github.com/vladmandic/human/blob/cf08ef892c65ed984b2367b59bcac41a35c310fe/src/face/angles.ts#L30>
